### ISSUE TITLE
Error: Insufficient data when trying to decode arguments of filter with no arguments

### ISSUE THREAD

**Issue by tmpolaczyk** [witnet/witnet-requests-js#57]
It looks like the toolkit assumes that every filter will have valid CBOR as args, but in this case the reducer has no arguments, and trying to decode `[]` as CBOR fails.

Command:

```
npx -p witnet-requests witnet-toolkit try-data-request --hex 0ac201122b0801122468747470733a2f2f626c6f636b636861696e2e696e666f2f712f6c6174657374686173681a018012390801122868747470733a2f2f6170692d722e626974636f696e636861696e2e636f6d2f76312f7374617475731a0b8218778218676468617368124c0801122868747470733a2f2f6170692e626c6f636b63686169722e636f6d2f626974636f696e2f73746174731a1e83187782186664646174618218676f626573745f626c6f636b5f686173681a02100222060a020808100210e8071803200a2833308094ebdc03
```

Output:

```
(node:2130410) UnhandledPromiseRejectionWarning: Error: Insufficient data
    at Object.decodeFirstSync (/home/tomasz/.npm/_npx/2130297/lib/node_modules/witnet-requests/node_modules/cbor/lib/decoder.js:218:15)
    at /home/tomasz/.npm/_npx/2130297/lib/node_modules/witnet-requests/dist/witnet/toolkit.js:352:28
    at Array.map (<anonymous>)
    at decodeScriptsAndArguments (/home/tomasz/.npm/_npx/2130297/lib/node_modules/witnet-requests/dist/witnet/toolkit.js:351:49)
    at _callee7$ (/home/tomasz/.npm/_npx/2130297/lib/node_modules/witnet-requests/dist/witnet/toolkit.js:426:35)
    at tryCatch (/home/tomasz/.npm/_npx/2130297/lib/node_modules/witnet-requests/node_modules/regenerator-runtime/runtime.js:63:40)
    at Generator.invoke [as _invoke] (/home/tomasz/.npm/_npx/2130297/lib/node_modules/witnet-requests/node_modules/regenerator-runtime/runtime.js:294:22)
    at Generator.next (/home/tomasz/.npm/_npx/2130297/lib/node_modules/witnet-requests/node_modules/regenerator-runtime/runtime.js:119:21)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:2130410) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:2130410) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

This is the line that causes the error:

https://github.com/witnet/witnet-requests-js/blob/2611da21d3c35e1ba4bb815805a6aa95008d53b6/src/witnet/toolkit.js#L212

This is the erroneous data request, which uses `"filters":[{"op":8,"args":[]}]`.

```
{"jsonrpc":"2.0","method":"sendRequest","id":"1","params":{"dro":{"data_request":{"time_lock":0,"retrieve":[{"kind":"HTTP-GET","url":"https://blockchain.info/q/latesthash","script":[128]},{"kind":"HTTP-GET","url":"https://api-r.bitcoinchain.com/v1/status","script":[130,24,119,130,24,103,100,104,97,115,104]},{"kind":"HTTP-GET","url":"https://api.blockchair.com/bitcoin/stats","script":[131,24,119,130,24,102,100,100,97,116,97,130,24,103,111,98,101,115,116,95,98,108,111,99,107,95,104,97,115,104]}],"aggregate":{"filters":[],"reducer":2},"tally":{"filters":[{"op":8,"args":[]}],"reducer":2}},"witness_reward":1000,"witnesses":3,"commit_and_reveal_fee":10,"min_consensus_percentage":51,"collateral":1000000000},"fee":0}}
```

**Comment by Tommytrg** [comment]
Is this still a problem or was it solved in #66?

**Comment by tmpolaczyk** [comment]
Running the `npx` command from the issue description does not work for me, but the error is different:

```
npm ERR! code ENOENT
npm ERR! syscall chmod
npm ERR! path /home/tomasz/.npm/_npx/952028/lib/node_modules/witnet-requests/dist/ethereum/truffle/rad2sol.js
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, chmod '/home/tomasz/.npm/_npx/952028/lib/node_modules/witnet-requests/dist/ethereum/truffle/rad2sol.js'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/tomasz/.npm/_logs/2022-08-05T12_32_26_224Z-debug.log
La instalación de [ 'witnet-requests@latest' ] fracasó con código 254
```

Regardless of that, I wouldn't consider this issue fixed until there is a test for that case somewhere in the repo.

**Comment by Tommytrg** [comment]
> Running the `npx` command from the issue description does not work for me, but the error is different:
> 
> ```
> npm ERR! code ENOENT
> npm ERR! syscall chmod
> npm ERR! path /home/tomasz/.npm/_npx/952028/lib/node_modules/witnet-requests/dist/ethereum/truffle/rad2sol.js
> npm ERR! errno -2
> npm ERR! enoent ENOENT: no such file or directory, chmod '/home/tomasz/.npm/_npx/952028/lib/node_modules/witnet-requests/dist/ethereum/truffle/rad2sol.js'
> npm ERR! enoent This is related to npm not being able to find a file.
> npm ERR! enoent 
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /home/tomasz/.npm/_logs/2022-08-05T12_32_26_224Z-debug.log
> La instalación de [ 'witnet-requests@latest' ] fracasó con código 254
> ```
> 
> Regardless of that, I wouldn't consider this issue fixed until there is a test for that case somewhere in the repo.

@guidiaz is this the same error that you had a few days ago?

**Comment by aesedepece** [comment]
> Is this still a problem or was it solved in #66?

This bug seems to be gone on master. That part has also been totally refactored in #73.
The error that @tmpolaczyk is getting now is fixed by #72.
